src/torcontrol.cpp

```diff
@@ -373,9 +373,18 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
     if (reply.code == 250) {
         LogPrint(BCLog::TOR, "tor: Authentication successful\n");
 
-        // Now that we know Tor is running setup the proxy for onion addresses
-        // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        // Now that we know Tor is running, possibly set the proxy for onion addresses.
+        bool allow_outbound_onion{true};
+        if (gArgs.IsArgSet("-onlynet")) {
+            allow_outbound_onion = false;
+            for (const auto& net : gArgs.GetArgs("-onlynet")) {
+                if (ParseNetwork(net) == NET_ONION) {
+                    allow_outbound_onion = true;
+                    break;
+                }
+            }
+        }
+        if (gArgs.GetArg("-onion", "") == "" && allow_outbound_onion) {
             CService resolved(LookupNumeric("127.0.0.1", 9050));
```

Comment:
This is normally 9151 on Windows so can we use #ifdef here?

```suggestion
#ifdef WIN32 
            CService resolved(LookupNumeric("127.0.0.1", 9150));
#else
            CService resolved(LookupNumeric("127.0.0.1", 9050));
#endif
```

Reply:
Yes, ouf of scope of this PR. This belongs to https://github.com/bitcoin/bitcoin/pull/15423, I hope that PR moves forward.